src/types.ts

```diff
@@ -295,16 +295,15 @@ export type StatesDefinition<
 };
 
 export type TransitionsConfig<TContext, TEvent extends EventObject> = {
-  [K in TEvent['type'] | BuiltInEvent<TEvent>['type']]?:
+  [K in TEvent['type'] | BuiltInEvent<TEvent>['type']]?: SingleOrArray<
```

Comment:
We should not have `SingleOrArray` here since the only array should be an array of `TransitionConfig` objects. In non-TS envs, `string[]` will produce the warning, but in TS envs it will cause a compilation error, which is good.

Reply:
I was thinking about this and came to the conclusion that it's better here to fail at runtime rather than when compiling. TS errors can't be customized - a runtime error here gives way more context to the user, so I would argue it's actually more helpful.

This change also allows for such config - https://github.com/davidkpiano/xstate/pull/577/files#diff-6112fd8214555eb54874e002969533ebR355 and this seems to make sense to me. We shouldn't forbid a simple form of a fallback target.